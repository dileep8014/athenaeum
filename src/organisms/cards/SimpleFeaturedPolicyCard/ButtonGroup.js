import React from 'react';
import PropTypes from 'prop-types';

import Button from 'atoms/Button';
import LinkWrapper from 'atoms/LinkWrapper';
import Spacer from 'atoms/Spacer';

import styles from './featured_policy_card.module.scss';

function ButtonGroup(props) {
  const {
    onContinue,
    onDetails,
    onCompare,
    continueCTAText,
    detailsCTAText,
  } = props;

  return (
    <div className={styles['button-group']}>
      <Button
        onClick={onContinue}
      >
        {continueCTAText}
      </Button>

      <Spacer spacer={2} />
      {
        onDetails &&
          <LinkWrapper
            onClick={onDetails}
            className={styles['details-link']}
          >
            {detailsCTAText}
          </LinkWrapper>

      }

      <Button outline onClick={onCompare} >
        Compare
      </Button>
    </div>
  );
}

ButtonGroup.propTypes = {
  onContinue: PropTypes.func.isRequired,
  onDetails: PropTypes.func,
  onCompare: PropTypes.func,
  continueCTAText: PropTypes.string,
  detailsCTAText: PropTypes.string,
};

ButtonGroup.defaultProps = {
  continueCTAText: 'Continue',
  detailsCTAText: 'Details',
};

export default ButtonGroup;
