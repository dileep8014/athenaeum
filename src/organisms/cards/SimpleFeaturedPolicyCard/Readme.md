### SimpleFeaturedPolicyCard Example:

```jsx
    <SimpleFeaturedPolicyCard
      carrierLogo={(
        <img
          src='http://res.cloudinary.com/policygenius/image/upload/v1500328467/mutual-of-omaha-dark.svg'
          alt='hello'
        />
      )}
      premium={{
        price: 26.73,
        format: 'mo',
        defaultText: 'Quote available from a PolicyGenius expert'
      }}
      onContinue={() => alert('continue button clicked')}
      compareCheckbox={{
        onCompare: (e) => setState({ compare: e.target.checked }),
        compareSelected: state.compare,
        name: 'unique-name'
      }}
    />
```
