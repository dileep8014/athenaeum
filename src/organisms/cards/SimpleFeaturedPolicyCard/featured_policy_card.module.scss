$card-border: 1px solid #dfdfdf;
$header-fill: color('neutral-5');
$radius: rem-calc(8px);
$details-link-color: #4a4a4a;
$details-link-border: 1px solid #b2b2b2;

.featured-policy-card {
  border: $card-border;
  background-color: color('primary-2');
  display: flex;
  flex-direction: column;
  height: 100%;

  &.selected {
    border: 1px solid color('secondary-2');
  }

  .details-link {
    @include typography-7(false);

    padding-bottom: 2px;
    color: $details-link-color;
    border-bottom: $details-link-border;
    text-decoration: none;
    display: none;

    &:hover {
      border-color: $details-link-color;
    }
  }
}

.content {
  padding: rem-calc(36px) rem-calc(24px);
  background-color: color('primary-2');
  border-radius: $radius;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  flex-grow: 1;
}

.premium {
  text-align: center;
}

.carrier-info {
  text-align: center;
}

.carrier-logo {
  align-items: center;
  display: flex;
  height: rem-calc(60px);
  justify-content: center;
  max-width: rem-calc(170px);

  img {
    max-width: 100%;
    max-height: 100%;
  }
}

.button-group {
  width: 100%;
  text-align: center;
  margin-top: auto;
}

.checkbox-wrapper {
  align-items: center;
  display: flex;
  flex-direction: column;

  .checkbox-field {
    margin-left: rem-calc(12px);
  }
}

@media #{$small-only} {
  .carrier-logo {
    max-width: rem-calc(150px);

    img {
      max-width: rem-calc(150px);
    }
  }
}

@media #{$medium-up} {
  .featured-policy-card {
    .details-link {
      display: inline-block;
    }
  }
}
