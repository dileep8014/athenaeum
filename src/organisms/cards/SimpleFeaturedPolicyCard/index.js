import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import accounting from 'accounting';

import Text from 'atoms/Text';
import Spacer from 'atoms/Spacer';

import styles from './featured_policy_card.module.scss';
import ButtonGroup from './ButtonGroup';

function SimpleFeaturedPolicyCard(props) {
  const {
    className,
    premium,
    carrierLogo,
    onContinue,
    onCompare,
  } = props;

  const classes = [
    styles['featured-policy-card'],
    className,
  ];

  const formattedPremium = accounting.formatMoney(premium.price);

  return (
    <div className={classnames(...classes)}>
      <div className={styles['content']}>
        { premium.price ?
          <div className={styles['premium']}>
            <Text
              type={3}
              font='a'
            >
              {formattedPremium} <Text tag='span' type={11} font='a' spaced color='neutral-2'>{`/${premium.format.toUpperCase()}`}</Text>
            </Text>
          </div>
          :
          <Text type={7} font='a'>{premium.defaultText}</Text>
        }

        <Spacer size={24} />

        <div className={styles['carrier-logo']}>
          { carrierLogo }
        </div>

        <Spacer size={36} />

        <ButtonGroup
          onContinue={onContinue}
          onCompare={onCompare}
        />
      </div>
    </div>
  );
}


SimpleFeaturedPolicyCard.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * Array of nodes to display in header
   */
  header: PropTypes.arrayOf(PropTypes.node),

  /**
   * Displays premium value
   *
   * `defaultText` replaces `price` is `price` is not available
   */
  premium: PropTypes.shape({
    price: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    format: PropTypes.string,
    defaultText: PropTypes.string,
  }),

  /**
   * Displays discount message below premium
   */
  discount: PropTypes.node,

  /**
   * Image for carrier logo
   */
  carrierLogo: PropTypes.node,

  /**
   * Text displayed for the main action button
   */
  continueCTAText: PropTypes.string,

  /**
   * Function supplied to continue CTA button
   */
  onContinue: PropTypes.func,

  /**
   * Function supplied to details link below CTA
   */
  onDetails: PropTypes.func,

  /**
   * Function supplied to compare CTA on mobile only
   */
  onCompare: PropTypes.func,

  /**
   * Supplies information about policy to card. Examples would include type, financial strength or total customers
   */
  information: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]).isRequired,
    hoverMessage: PropTypes.node.isRequired,
  })),
};

SimpleFeaturedPolicyCard.defaultProps = {
  continueCTAText: 'Continue',
};

export default SimpleFeaturedPolicyCard;
