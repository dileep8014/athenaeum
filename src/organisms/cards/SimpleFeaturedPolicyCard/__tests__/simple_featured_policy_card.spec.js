import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import SimpleFeaturedPolicyCard from 'organisms/cards/SimpleFeaturedPolicyCard';

describe('<SimpleFeaturedPolicyCard />', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      carrierLogo: <div />,
      premium: {
        price: 19.8,
        format: 'mo'
      },
      onDetails: jest.fn(),
      compareCheckbox: {
        onCompare: jest.fn(),
        compareSelected: false,
        name: 'a_unique_name',
      }
    };
  });

  it('renders correctly', () => {
    const wrapper = mount(<SimpleFeaturedPolicyCard {...defaultProps} />);

    expect(wrapper.type()).toEqual(SimpleFeaturedPolicyCard);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<SimpleFeaturedPolicyCard {...defaultProps} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
