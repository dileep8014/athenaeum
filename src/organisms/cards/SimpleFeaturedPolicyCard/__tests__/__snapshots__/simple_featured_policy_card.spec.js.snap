// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleFeaturedPolicyCard /> renders correctly 1`] = `
<div
  className="featured-policy-card"
>
  <div
    className="content"
  >
    <div
      className="premium"
    >
      <p
        className="primary-3 type-a-3-bold inherit"
      >
        $19.80
         
        <span
          className="neutral-2 spaced type-a-11-bold inherit"
        >
          /MO
        </span>
      </p>
    </div>
    <div
      className="size-24"
    />
    <div
      className="carrier-logo"
    >
      <div />
    </div>
    <div
      className="size-36"
    />
    <div
      className="button-group"
    >
      <button
        className="button"
        disabled={undefined}
        onClick={undefined}
        type="button"
      >
        Continue
      </button>
      <div
        className="top-2"
      />
      <a
        className="link neutral-1 details-link"
        onClick={[Function]}
      >
        Details
      </a>
      <button
        className="button outline"
        disabled={undefined}
        onClick={undefined}
        type="button"
      >
        Compare
      </button>
    </div>
  </div>
</div>
`;
